### PR TITLE
SAAS-393 execute e2e at minikube

### PR DESCRIPTION
**Pull request by yurkovychv**
updated `ci.yml` workflow and added e2e execution job after Building Docker Image. e2e.yml workflow executes tests with `latest` tags nightly

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -110,3 +110,10 @@ jobs:
 
       - name: Push Docker image
         run: make docker-push
+
+      - name: Trigger E2E execution with tag "${{ env.DOCKER_TAG }}"
+        uses: percona-platform/repository-dispatch@v1
```
Comment (AlekSi) [inline .github/workflows/ci.yml:115]: Why not use https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#workflow_dispatch directly?

Comment (AlekSi) [reply .github/workflows/ci.yml:115]: Err, that's confusing :) What I really wanted to ask is why use repository_dispatch instead of workflow_dispatch?

Comment (yurkovychv) [reply .github/workflows/ci.yml:115]: There is no particular reason. Will try with workflow_dispatch

**Review by AlekSi — CHANGES_REQUESTED** [review]
We can't just clone infra repository there and copy&paste all setup instructions there – it would be very hard to keep them in sync, especially once we copy&paste in other repos too.

We need find solution that will run tests in the context of infra repo or separate cicd repo. We talked about workflow_dispatch and repository_dispatch – was there any follow up?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (AlekSi) [reply .github/workflows/ci.yml]: Why that was abandoned?

**Comment by yurkovychv** [comment]
I need a spike for that, I was testing it with workflow_dispatch. But it does not affect PR checks (if tests failed a PR does not indicate that, and it is not clear which exact PR triggered that workflow). It just says `Manually run by percona-robot`. The best way would be to have 1 workflow in `cicd` repo. But again we need to find a way to detect which PR triggered it (without reading logs) and indicate it in the original PR.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (yurkovychv) [reply .github/workflows/ci.yml]: I replied [here](https://github.com/percona-platform/saas-ui/pull/44#issuecomment-720983995)

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -29,51 +93,42 @@ jobs:
...
-          start: yarn start
-          wait-on: 'http://localhost:3000'
-          wait-on-timeout: 120
+          working-directory: saas-ui/packages/platform-ui
+          command: yarn cy:run
 
       - name: Generating the report
-        if: success() || failure()
-        working-directory: packages/platform-ui
+        if: ${{ always() }}
```
Comment (nicolalamacchia) [inline .github/workflows/e2e.yml:110]: Why is this needed?

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -29,51 +93,42 @@ jobs:
         id: cache
         with:
           path: |
-            node_modules
-            */*/node_modules
+            saas-ui/node_modules
+            */*/*/node_modules
```
Comment (nicolalamacchia) [inline .github/workflows/e2e.yml:97]: would `**/nodemodules` work here?

**Review by AlekSi — APPROVED** [review]

File: .github/workflows/e2e.yml (hunk 2 above)
Comment (AlekSi) [reply .github/workflows/e2e.yml:110]: It can also be `canceled`, and it is different from being failed

Comment (yurkovychv) [reply .github/workflows/e2e.yml:110]: Yes, we need this step to be executed anyway

File: .github/workflows/e2e.yml (hunk 3 above)
Comment (yurkovychv) [reply .github/workflows/e2e.yml:97]: not sure, should I try It now? Anyway, this workflow will be moved to another repo and updated soon.

**Review by nicolalamacchia — APPROVED** [review]

**Pull request merged**